app/bundles/CampaignBundle/Entity/CampaignRepository.php

```diff
@@ -637,4 +637,37 @@ public function fetchEmailIdsById($id)
 
         return $return;
     }
+
+    /**
+     * @param \Doctrine\Common\Collections\ArrayCollection<string, mixed> $contacts
+     * @param \Mautic\CampaignBundle\Entity\Campaign                      $campaign
+     *
+     * @return void
+     */
+    public function unscheduleContacts($contacts, $campaign)
+    {
+        $campaignId = $campaign->getId();
+        $contactIds = [];
+        foreach ($contacts as $contact) {
+            $contactIds[] = $contact->getId();
+        }
```

Comment:
I'm just geeking out here, but another option how to write this is
```suggestion
        $contactIds = $contacts->map(fn (Lead $contact) => return (int) $contact->getId());
```